File: Java/Structural/CompositePattern.java
package Structural;

import java.util.ArrayList;
import java.util.List;

// Component
interface Graphic {
    void draw();
}

// Leaf
class Circle implements Graphic {
    @Override
    public void draw() {
        System.out.println("Drawing Circle");
    }
}

class Rectangle implements Graphic {
    @Override
    public void draw() {
        System.out.println("Drawing Rectangle");
    }
}

// Composite
class CompositeGraphic implements Graphic {
    private List<Graphic> graphics = new ArrayList<>();
    
    public void add(Graphic graphic) {
        graphics.add(graphic);
    }
    
    public void remove(Graphic graphic) {
        graphics.remove(graphic);
    }
    
    @Override
    public void draw() {
        for(Graphic graphic : graphics) {
            graphic.draw();
        }
    }
}

File: Python/Structural/CompositePattern.py
# Component
class Graphic:
    def draw(self):
        pass

# Leaf
class Circle(Graphic):
    def draw(self):
        print("Drawing Circle")

class Rectangle(Graphic):
    def draw(self):
        print("Drawing Rectangle")

# Composite
class CompositeGraphic(Graphic):
    def __init__(self):
        self.graphics = []

    def add(self, graphic):
        self.graphics.append(graphic)

    def draw(self):
        for graphic in self.graphics:
            graphic.draw()

# Client usage
circle = Circle()
rectangle = Rectangle()
composite_graphic = CompositeGraphic()

composite_graphic.add(circle)
composite_graphic.add(rectangle)

composite_graphic.draw()  # Drawing Circle, Drawing Rectangle
